kernel.py

```diff
@@ -105,6 +97,22 @@ def __init__(self, connection_filename="kernel.json", connection_fn_with_pid=Tru
             logger.addHandler(logging.StreamHandler(sys.stdout))
         self._logger = logger
 
+    def _get_connection_filename(self, connection_filename, connection_fn_with_pid):
+        """Craft connection file path"""
+        if connection_fn_with_pid and connection_filename is None:
+            self._should_reduce_filename = True
```

Comment:
Such a function should not have side effects, like setting this attrib. You would not expect that from the function name.
But maybe easier, just do not create such a separate function at all.

Reply:
Also, this is wrong here. This should come after the `os.path.join` part. Only for that case, we should set `_should_reduce_filename = True`.

Reply:
> Also, this is wrong here. This should come after the `os.path.join` part. Only for that case, we should set `_should_reduce_filename = True`.

Well seen: the user may not have jupyter (and its path and its connection file searching routine). It was a (well hidden) error.
Fix: 4914afb70bc